csharp/src/Apache.Arrow/Ipc/ArrowStreamReader.cs

```diff
@@ -31,26 +31,41 @@ public class ArrowStreamReader : IArrowReader, IDisposable
         public Schema Schema => _implementation.Schema;
 
         public ArrowStreamReader(Stream stream)
-            : this(stream, allocator: null, leaveOpen: false)
+            : this(stream, allocator: null, compressionCodecFactory: null, leaveOpen: false)
         {
         }
 
         public ArrowStreamReader(Stream stream, MemoryAllocator allocator)
-            : this(stream, allocator, leaveOpen: false)
+            : this(stream, allocator, compressionCodecFactory: null, leaveOpen: false)
+        {
+        }
+
+        public ArrowStreamReader(Stream stream, ICompressionCodecFactory compressionCodecFactory)
+            : this(stream, allocator: null, compressionCodecFactory, leaveOpen: false)
         {
         }
 
         public ArrowStreamReader(Stream stream, bool leaveOpen)
-            : this(stream, allocator: null, leaveOpen)
+            : this(stream, allocator: null, compressionCodecFactory: null, leaveOpen)
         {
         }
 
         public ArrowStreamReader(Stream stream, MemoryAllocator allocator, bool leaveOpen)
+            : this(stream, allocator, compressionCodecFactory: null, leaveOpen)
+        {
+        }
+
+        public ArrowStreamReader(Stream stream, ICompressionCodecFactory compressionCodecFactory, bool leaveOpen)
+            : this(stream, allocator: null, compressionCodecFactory, leaveOpen)
+        {
+        }
+
+        public ArrowStreamReader(Stream stream, MemoryAllocator allocator, ICompressionCodecFactory compressionCodecFactory, bool leaveOpen)
         {
             if (stream == null)
                 throw new ArgumentNullException(nameof(stream));
 
-            _implementation = new ArrowStreamReaderImplementation(stream, allocator, leaveOpen);
+            _implementation = new ArrowStreamReaderImplementation(stream, allocator, compressionCodecFactory, leaveOpen);
         }
 
         public ArrowStreamReader(ReadOnlyMemory<byte> buffer)
```

Comment:
Why not add decoder support for ArrowMemoryReaderImplementation?

Reply:
That wasn't intentional, I'd missed that there was a separate implementation for reading from a memory buffer. I'll make a PR to fix this.

Reply:
PR opened at #34108